src/assets/scripts/client/aircraft/AircraftInstanceModel.js

```diff
@@ -377,19 +377,39 @@ export default class AircraftInstanceModel {
     }
 
     /**
+     * Returns a true value if there is a match from the callsignToMatch
+     *
      * @for AircraftInstanceModel
      * @method matchCallsign
      * @param callsign {string}
      */
     matchCallsign(callsignToMatch) {
-        if (callsignToMatch === '*') {
+        const shouldMatchAnyCallsign = callsignToMatch === '*';
+         // checks to see if the given call sign matches the airline Id + callsign format
+        if (shouldMatchAnyCallsign || (this.airlineId.toUpperCase() + callsignToMatch.toUpperCase() === this.callsign)) {
             return true;
         }
 
+        // Checks to see if the given callsign matches only the callsign since callsign numbers should be unique
         return _isEqual(callsignToMatch.toUpperCase(), this.callsign);
```

Comment:
does this method work the same way when used with any of the following:
- `AAL123`
- `123`
- `AaL123`